src/components/Recipes/index.tsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react'
+import { Link, useLocation } from 'react-router-dom'
+
+import { fetchRecipes } from 'api'
+import { IRecipe } from 'types'
+import Button from 'components/common/Button'
+
+import styles from './Recipes.module.scss'
+
+function Recipes() {
+  const [recipes, setRecipes] = useState([])
+  const location = useLocation()
+  const query = location.state.query
```

Comment:
```suggestion
  const { query } = location.state
```

does it work when someone directly open this page in browser? without submitting Search form?

Reply:
It didn't work then, I'm redirecting to search page if query isn't provided